### PR TITLE
Enabled wowinterface for addon matching

### PR DESCRIPTION
**Pull request by jukx**
Made wowinterface matching work with both the search overview and redirection to addon page. Before nothing worked.

### CODE REVIEW

File: sites.lua [hunk 1]
```diff
@@ -229,18 +229,44 @@ parseSearchResults["wowace"] = parseSearchResults["curseforge"]
...
-		local author = tr:match('<a href="/forums/member%.php[^"]+">(.-)</a>')
-		local updated = tr:match('<td align="center" class="alt%d">(%d%d%-%d%d%-%d%d)</td>')
-		if updated then
-			local m, d, y = tr:match("(%d%d)%-(%d%d)%-(%d%d)")
-			if m and d and y then
-				updated = getTimestamp("20"..y, m, d)
-			end
-		end
-	end
+    cleanedHtml = cleanHTML(html)
```
Comment (Phanx) [inline sites.lua]: This should be a local variable, not global.

File: sites.lua [hunk 2]
```diff
@@ -229,18 +229,44 @@ parseSearchResults["wowace"] = parseSearchResults["curseforge"]
...
+		if am == "PM" then hh = tonumber(hh) + 12 end
+		local date = getTimestamp(y, m, d, hh, mm)
+
+        if id and name and author then
+            table.insert(t, {
+                site = site,
+                id = id,
+                name = name,
+                author = author,
+                date = tonumber(date),
```
Comment (Phanx) [inline sites.lua]: No need for `tonumber` here; the value returned by `getTimestamp` is already a number.

File: sites.lua [hunk 3]
```diff
@@ -229,18 +229,44 @@ parseSearchResults["wowace"] = parseSearchResults["curseforge"]
...
-			end
-		end
-	end
+    cleanedHtml = cleanHTML(html)
+    if cleanedHtml:match('>Addon Info.<') then
+        local id = cleanedHtml:match('/downloads/download(%d+)%-')
+        local name = cleanedHtml:match('<h1>(.-)</h1>')
+        local author = cleanedHtml:match('/forums/member%.php[^"]+"><b>(.-)</b>')
+		local d, m, y, hh, mm, am = cleanedHtml:match('<div id="safe">Updated: (%d+)%-(%d+)%-(%d+) (%d+):(%d+) ([AP]M)</div>')
+		if am == "PM" then hh = tonumber(hh) + 12 end
```
Comment (Phanx) [inline sites.lua]: Let's safety-check this and make sure the `match` from the previous line actually resulted in values before we try to manipulate the values.

File: sites.lua [hunk 4]
```diff
@@ -229,18 +229,44 @@ parseSearchResults["wowace"] = parseSearchResults["curseforge"]
...
+		local date = getTimestamp(y, m, d, hh, mm)
+
+        if id and name and author then
+            table.insert(t, {
+                site = site,
```
Comment (Phanx) [inline sites.lua]: Should be `site = "wowinterface",` after f2823de

File: sites.lua [hunk 5]
```diff
@@ -229,18 +229,44 @@ parseSearchResults["wowace"] = parseSearchResults["curseforge"]
...
+            local m, d, y = cleanedHtml:match('<td align="center" class="alt1">(%d+)%-(%d+)%-(%d+)</td>')
+            local date = getTimestamp(y, m, d) 
+
+            if id and name and author then
+                table.insert(t, {
+                    site = site,
+                    id = id,
+                    name = name,
+                    author = author,
+                    date = tonumber(date),
```
Comment (Phanx) [inline sites.lua]: Same as above; no need for `tonumber`.

**Review by Phanx — CHANGES_REQUESTED** [review]
Needs indentation cleanup throughout; use tabs, not spaces, and make sure levels are consistent (e.g. line 237 should be indented the same as 236 which is +1 indent from 233, but is currently indented less than both).

**Comment by jukx** [comment]
Issues should be fixed.

File: sites.lua (hunk 3 above)
Comment (jukx) [reply sites.lua]: now checking hh before modifying it

**Review by Phanx — APPROVED** [review]

**Pull request merged**